phoenix-core/src/main/java/org/apache/phoenix/jdbc/PhoenixStatement.java

```diff
@@ -417,11 +508,20 @@ private PhoenixResultSet executeQuery(final CompilableStatement stmt,
                                                     e.getSchemaName(), e.getTableName(), true)
                                             .wasUpdated()) {
                                         //TODO we can log retry count and error for debugging in LOG table
-                                        return executeQuery(stmt, false, queryLogger, noCommit);
+                                        return executeQuery(stmt, false, queryLogger, noCommit, validateLastDdlTimestamp);
                                     }
                                 }
                                 throw e;
-                            } catch (RuntimeException e) {
+                            } catch (StaleMetadataCacheException e) {
```

Comment:
I think you need to move this catch block inside where you are calling validateTimestamp otherwise the finally block will be executed twice when StaleMetadataCacheException is raised.